falcon/request.py

```diff
@@ -1028,6 +1059,22 @@ def _parse_form_urlencoded(self):
 
             self._params.update(extra_params)
 
+    def _parse_rfc_forwarded(self):
+        """rfc 7239"""
+        addr = []
+        for forwarded in self.env['HTTP_FORWARDED'].split(','):
+            for param in forwarded.split(';'):
+                param = param.strip().split('=', 1)
+                if len(param) == 1:
+                    continue
+                key, val = param
+                if key.lower() not in ('for', 'by'):
+                    # we only want for/by params
+                    continue
+                host, _ = parse_host(unquote_string(val))
```

Comment:
Well, I didn't notice the "SHOULD NOT" part of RFC 7230 before. I just followed the requirement of ABNF:

```
quoted-pair    = "\" ( HTAB / SP / VCHAR / obs-text )
```

Since `VCHAR` stands for "any visible USASCII character", according to the ABNF (only), it seems legal to generate a `quoted-string` with unnecessary `quoted-pair`, for example `"127\.0\.0\.1"`. That's the reason I introduced the `unquote_string()` function at first.

So there's a balance to be made between the robust and performance. What do you think @kgriffs?